goldenmaster/Plugins/ApiGear/Source/ApiGearOLink/Private/OLinkHostPrivate.cpp

```diff
@@ -0,0 +1,132 @@
+		break;
+	case ApiGear::ObjectLink::LogLevel::Error:
+		UE_LOG(LogApiGearOLinkHost, Error, TEXT("%s"), UTF8_TO_TCHAR(msg.c_str()));
+		break;
+	}
+}
+
+ApiGear::ObjectLink::WriteLogFunc logHostFunc()
+{
+	return [](ApiGear::ObjectLink::LogLevel level, const std::string& msg)
+	{ writeLogHost(level, msg); };
+}
+
+OLinkHostPrivate::OLinkHostPrivate(uint32 InPort)
+	: m_loggingDisabled(false)
+	, Port(InPort)
+{
+	UApiGearSettings* settings = GetMutableDefault<UApiGearSettings>();
+
+	Registry = MakeShared<ApiGear::ObjectLink::RemoteRegistry>();
+	Registry->onLog(logHostFunc());
+
+	if (settings->OLINK_AutoStart)
+	{
+		Start(Port);
+	}
+}
+
+OLinkHostPrivate::~OLinkHostPrivate()
+{
+	Stop();
+}
+
+bool OLinkHostPrivate::Start(uint32 InPort)
+{
+	writeLogHost(ApiGear::ObjectLink::LogLevel::Debug, "Apigear OLink Server starting up...");
+	Port = InPort;
+
+	FWebSocketClientConnectedCallBack CallBack;
+	CallBack.BindRaw(this, &OLinkHostPrivate::OnWebSocketClientConnected);
+
+	Server = FModuleManager::Get().LoadModuleChecked<IWebSocketNetworkingModule>(TEXT("WebSocketNetworking")).CreateServer();
+
+	if (!Server || !Server->Init(Port, CallBack))
+	{
+		static const std::string ErrorMsg = "Cannot start new OLink server";
+		writeLogHost(ApiGear::ObjectLink::LogLevel::Error, ErrorMsg);
+		Server.Reset();
+		return false;
+	}
+
+#if (ENGINE_MAJOR_VERSION >= 5)
+	TickerHandle = FTSTicker::GetCoreTicker().AddTicker(FTickerDelegate::CreateRaw(this, &OLinkHostPrivate::Tick));
+#else
+	TickerHandle = FTicker::GetCoreTicker().AddTicker(FTickerDelegate::CreateRaw(this, &OLinkHostPrivate::Tick));
+#endif
+	return true;
+}
+
+void OLinkHostPrivate::Stop()
+{
+	if (TickerHandle.IsValid())
+	{
+#if (ENGINE_MAJOR_VERSION >= 5)
+		FTSTicker::GetCoreTicker().RemoveTicker(TickerHandle);
+#else
+		FTicker::GetCoreTicker().RemoveTicker(TickerHandle);
+#endif
+
+		TickerHandle.Reset();
+	}
+	if (Server)
+	{
+		Server.Reset();
+	}
+}
+
+bool OLinkHostPrivate::Tick(float DeltaTime)
+{
+	Server->Tick();
+	return true;
+}
+
+void OLinkHostPrivate::OnWebSocketClientConnected(INetworkingWebSocket* Socket)
+{
+	checkf(Socket, TEXT("Socket was null while creating a new websocket connection."));
+
+	static const std::string Msg = "remote: new connection ";
+	writeLogHost(ApiGear::ObjectLink::LogLevel::Info, Msg + TCHAR_TO_UTF8(*Socket->RemoteEndPoint(true)));
+	TUniquePtr<FOLinkHostConnection> Connection = MakeUnique<FOLinkHostConnection>(Socket, *Registry, logHostFunc());
+
+	Connection->ConnectionClosedCallBack.BindRaw(this, &OLinkHostPrivate::OnConnectionClose);
+
+	ClientConnections.Add(MoveTemp(Connection));
+}
+
+void OLinkHostPrivate::OnConnectionClose(FOLinkHostConnection* Connection)
+{
+	ClientConnections.RemoveAllSwap([Connection](const TUniquePtr<FOLinkHostConnection>& InConnection)
+		{ return InConnection->IsConnection(Connection); });
+}
+
+uint32 OLinkHostPrivate::NumConnections() const
```

Comment:
MINOR I still don't like the function name(the "num" part), maybe one of those numberOfConnections,  connectionsSize, connectoinsCount

Reply:
done